x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/set_message_helpers.test.ts

```diff
@@ -72,4 +74,46 @@ describe('Flash Message Helpers', () => {
 
     expect(FlashMessagesLogic.values.messages).toEqual([]);
   });
+
+  describe('toast helpers', () => {
+    afterEach(() => {
+      FlashMessagesLogic.actions.clearToastMessages();
+    });
+
+    it('flashSuccessToast', () => {
+      flashSuccessToast({
+        id: 'successToastMessage',
+        title: 'You did a thing!',
+        toastLifeTimeMs: 500,
+      });
+
+      expect(FlashMessagesLogic.values.toastMessages).toEqual([
+        {
+          color: 'success',
+          iconType: 'check',
+          id: 'successToastMessage',
+          title: 'You did a thing!',
+          toastLifeTimeMs: 500,
+        },
+      ]);
+    });
+
+    it('flashErrorToast', () => {
+      flashErrorToast({
+        id: 'errorToastMessage',
+        title: 'Something went wrong',
+        text: "Here's some helpful advice on what to do",
+      });
```

Comment:
[Request for comments] Please let me know what you think of this API!

It is **not** similar to our current (simpler) flash messages helper, which looks more like this:
```js
setSuccessMessage('just a string');
```

But it is 1. more flexible, and 2. close to/essentially using EUI's `Toast` data structure (https://elastic.github.io/eui/#/display/toast).

If we really really wanted to, I **could** change this API to something like:

```js
flashSuccessToast('just a string')
```

and the helper attempts takes to take care of its own a unique ID (maybe based on a timestamp?) and disallows a custom description/timeout (unless we want to add a custom 2nd arg as an option... something like this?

```js
flashSuccessToast('main string message', { toastLifetimeMs: customMs });
```

I'm a little leery of that though... Another alternative could be to make a new fn like:

```js
flashSimpleSuccessToast('just a string')
```
in addition to `flashSuccessToast`.

Thoughts? Suggestions?

Reply:
My 2c: I think I'm leaning towards the 3rd extra `flashSimpleSuccessToast('')` helper, while leaving the full `flashSuccessToast({})` helper as-is for more complex use-cases.

Reply:
He, before I even got to your last comment, I had already copied `flashSimpleSuccessToast` to the clipboard to vote in favor of it. So I'm def on the same page 🎉 

One suggestion might be to simplify the default and give the complex one the odd name so that it aligns with flash messages. Also, something about having `flash` in toasts when we have flash messages seems weird so here is a alternative approach:

```JavaScript
setSuccessMessage('yay!');

// Simple toast
addToastMessage('yippie!')

// Complex
addAdvancedToastMessage('ponder your existence for an hour', { toastLifetimeMs, ONE_HOUR_MS });
```
Open to something other than `Advanced` but I think it would be a better developer experience to have the simple one be generic.

Reply:
I don't feel strongly either way about defaulting to `simple` vs `advanced` - will leave this convo open a bit longer in case @byronhulcher or @JasonStoltz have any preference there. If none, I'll make that change tomorrow.

Reply:
My suggestion ... one method, first position is required, second is optional parameters, then you don't have 2 different methods.

```
// Simple toast
flashSuccessToast('just a string')

// Advanced toast
flashSuccessToast('just a string' { toastLifetimeMs, 500, id: 'successToastMessage' });
```

Reply:
![thatsit](https://user-images.githubusercontent.com/1869731/113344033-31337c80-92f6-11eb-8008-f8084dbf04e9.gif)

Reply:
👍 Nice! I'm a +1!

Reply:
Okily dokily, here's the final implementation!

https://github.com/elastic/kibana/pull/95981/commits/f6e7ba567b73d3bb92d77f2a50ec68eb843403d3

Please leave feel free to add code feedback or suggestions. Thanks y'all for already greatly improving this PR! 🎉